server/libs/infra/src/search/typesense.repository.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  ISearchRepository,
+  SearchCollection,
+  SearchCollectionIndexStatus,
+  SearchFilter,
+  SearchResult,
+} from '@app/domain';
+import { Injectable, Logger } from '@nestjs/common';
+import { Client } from 'typesense';
+import { CollectionCreateSchema } from 'typesense/lib/Typesense/Collections';
+import { DocumentSchema, SearchResponse } from 'typesense/lib/Typesense/Documents';
+import { AlbumEntity, AssetEntity } from '../db';
+import { albumSchema } from './schemas/album.schema';
+import { assetSchema } from './schemas/asset.schema';
+
+const schemaMap: Record<SearchCollection, CollectionCreateSchema> = {
+  [SearchCollection.ASSETS]: assetSchema,
+  [SearchCollection.ALBUMS]: albumSchema,
+};
+
+const schemas = Object.entries(schemaMap) as [SearchCollection, CollectionCreateSchema][];
+
+// TODO: look at using an external file for advanced configuration
+export const typesense = new Client({
+  nodes: [
+    {
+      host: process.env.TYPESENSE_HOST || 'typesense',
+      port: Number(process.env.TYPESENSE_PORT) || 8108,
+      protocol: process.env.TYPESENSE_PROTOCOL || 'http',
+    },
+  ],
+  apiKey: process.env.TYPESENSE_API_KEY || 'should-not-be-used',
```

Comment:
I'm confused about this `'should-not-be-used'` default. Is it just a dev placeholder?

Reply:
API key is required and an env variable had a potentially undefined type. Will probably fix that differently before this is merged.

Reply:
Should we fix this now?

Reply:
Yeah, I'll fix it later tonight and clean up the PR for review.